pkg/syncer/generic.go

```diff
@@ -40,8 +41,8 @@ func (s *patcher) ApplyPatches(ctx context.Context, fromObj, toObj client.Object
 	}
 	toObjCopied := toObjBase.DeepCopy()
 
-	// apply patches on from object
-	err = patches.ApplyPatches(toObjCopied, toObj, patchesConfig, reversePatchesConfig, nameResolver)
+	// apply patches on toObjCopied
+	err = patches.ApplyPatches(toObjCopied, fromObj, patchesConfig, reversePatchesConfig, nameResolver)
```

Comment:
why do we want to do this?

Reply:
this is needed in cases where we want to populated controller applied default back into the virtual obj. i.e.
`vObj.field1 = pObj.field1`.

Currently the behavior of `copyFromObject` is like `vObj.field1 = vObj.field2`

But consider the scenario where
```
vobj
=====
spec.template.timeoutSeconds: nil (unset)
|
|
v
-------------------------------------------
pobj
spec.template.timeoutSeconds: 300 ----------> set by controller to default
```

I would want to (expect that) `copyFromObject` in `reversePatches` as defined
```
                  - op: copyFromObject
                    fromPath: spec.template.spec.timeoutSeconds
                    path: spec.template.spec.timeoutSeconds
                    conditions:
                      - empty: true
                        path: spec.template.spec.timeoutSeconds
```

would copy across physical <-> virtual boundary, but since both the args to `ApplyPatches` originated from same source obj, it was not happening